### DISCUSSION TITLE
[Meta] Screenshots

### DISCUSSION THREAD

**Discussion by bbinto** [mozilla-mobile/fenix#20594] [Ideas and Feature Requests]
Requirements (user story to be written)
- screenshot saved to your photos folder
- screenshot saved to your Fenix library
- options to save whole page, viewed area, a selection
- URL, title of image properties
- Integration screenshots

A-C: https://github.com/mozilla-mobile/android-components/issues/1826

**Comment by bbinto** [comment]
UX issue: https://github.com/mozilla-mobile/fenix/issues/183

**Comment by bbinto** [comment]
Need to confirm with GeckoView

**Comment by cpeterso** [comment]
GV bugs for taking screenshots of web content:
* viewport: https://bugzilla.mozilla.org/show_bug.cgi?id=1462018
* full page: https://bugzilla.mozilla.org/show_bug.cgi?id=1530967

**Comment by bbinto** [comment]
@vesta0 - I'm considering dropping this for MVP partially after talking to GV engineering manager and workload vs other priorities. Thoughts?

**Comment by cpeterso** [comment]
GV support for taking viewport screenshots (for page thumbnails in tab switcher UI) has already landed ([bug 1462018](https://bugzilla.mozilla.org/show_bug.cgi?id=1462018)). Full page shots ([bug 1530967](https://bugzilla.mozilla.org/show_bug.cgi?id=1530967)) is still planned but it will be more work.

**Comment by bbinto** [comment]
As discussed with @vesta0 - we could start with supporting viewport screenshot support for MVP if possible, but it should not be a MUST criteria. Keeping it on the board as a "could" seems reasonable.

**Comment by cpeterso** [comment]
> As discussed with @vesta0 - we could start with supporting viewport screenshot support for MVP if possible, but it should not be a MUST criteria. Keeping it on the board as a "could" seems reasonable.

GV has support for viewport screenshots (for page thumbnails in tab switcher UI) has landed and is ready for the AC team to start using (android-components#498).

The GV API for full page shots ([bug 1530967](https://bugzil.la/1530967)) is still planned but it will be more work.

**Comment by bbinto** [comment]
Considering de-scoping this for MVP as there is no real additional / new user value than using the current screenshot device functionality, this also came out of some research findings lately. @vesta0 - do you agree?

**Comment by JoGarWeb** [comment]
Now this is forgotten 😭💔🥺
